apps/photos/src/services/collectionService.ts

```diff
@@ -300,6 +300,40 @@ export const getCollectionLatestFiles = (
     return latestFiles;
 };
 
+export const getCollectionCoverFiles = (
+    user: User,
+    files: EnteFile[],
+    archivedCollections: Set<number>,
+    collections: Collection[]
+): CollectionToFileMap => {
+    const coverFiles = new Map<number, EnteFile>();
+
+    const collectionMap = new Map<number, Collection>();
+    collections.forEach((collection) => {
+        collectionMap.set(collection.id, collection);
+    });
+
+    files.forEach((file) => {
+        const collection = collectionMap.get(file.collectionID);
+        if (collection.pubMagicMetadata?.data?.coverID === file.id) {
+            coverFiles.set(file.collectionID, file);
+        } else if (collection.pubMagicMetadata?.data?.asc) {
+            coverFiles.set(file.collectionID, file);
+        } else if (!coverFiles.has(file.collectionID)) {
```

Comment:
If first file is cover and next is not, the else if logic will override the cover?

Reply:
It should not. 
 It checks if `coverFile` is `not` set.